includes/abstracts/abstract-wc-order.php

```diff
@@ -769,6 +770,23 @@ public function get_items( $types = 'line_item' ) {
 		return apply_filters( 'woocommerce_order_get_items', $items, $this, $types );
 	}
 
+	/**
+	 * Return array of values for calculations.
+	 *
+	 * @param string $field Field name to return.
+	 *
+	 * @return array Array of values.
+	 */
+	protected function get_values_for_total( $field ) {
```

Comment:
Perhaps a more general helper function would be better instead of having to duplicate this function here and in WC_Cart_Totals. I like the use of traits but it seems it defeats the purpose if we are overriding it with custom versions anyway in the different classes. There should be a way to make the trait function support both scenarios if we pass in the item/s

I am trying to avoid us having code in different places that are trying to achieve the same end goal, especially when it comes to taxes and calculations having duplicate functions is what is causing the current discrepancies we see.

Reply:
Refactored Trait class to don't this abstract method. This is a bit different from what you suggested but now Trait only contains functionality which is explicitly related to calculating totals and nothing else. I did not move WC_Order and WC_Cart specific logic inside the trait because then it was not looking cleaner, and we may end up using this logic in some other place as well, so probably better to keep specific code in that class. Wdyt?